ci-operator/step-registry/rosa/cluster/upgrade/hcp/rosa-cluster-upgrade-hcp-commands.sh

```diff
@@ -0,0 +1,226 @@
+  rosa list machinepool -c $cluster_id
+
+  up_version=$1
+  mp_id_list=$(rosa list machinepool -c $cluster_id -o json | jq -r ".[].id")
+  for mp_id in $mp_id_list; do
+    log "rosa upgrade machinepool $mp_id -y -c $cluster_id --version $up_version"
+    rosa upgrade machinepool $mp_id -y -c $cluster_id --version $up_version
+  done
+
+  for mp_id in $mp_id_list; do
+    start_time=$(date +"%s")
+    while true; do
+        sleep 120
+        log "Wait for the node upgrading for the machinepool $mp_id finished ..."
+        node_version=$(rosa list machinepool -c $cluster_id -o json | jq -r --arg k $mp_id '.[] | select(.id==$k) .version.id')
+        if [[ "$node_version" =~ ${up_version}- ]]; then
+          record_cluster "timers.machinset_upgrade" "${mp_id}" $(( $(date +"%s") - "${start_time}" ))
+          log "Upgrade the machinepool $mp_id successfully to the openshift version $up_version after $(( $(date +"%s") - ${start_time} )) seconds"
+          break
+        fi
+
+        if (( $(date +"%s") - $start_time >= $NODE_UPGRADE_TIMEOUT )); then
+          log "error: Timed out while waiting for the machinepool upgrading to be ready"
+          rosa list machinepool -c $cluster_id
+          exit 1
+        fi
+    done
+  done
+}
+
+
+read_profile_file() {
+  local file="${1}"
+  if [[ -f "${CLUSTER_PROFILE_DIR}/${file}" ]]; then
+    cat "${CLUSTER_PROFILE_DIR}/${file}"
+  fi
+}
+
+# Log in
+SSO_CLIENT_ID=$(read_profile_file "sso-client-id")
+SSO_CLIENT_SECRET=$(read_profile_file "sso-client-secret")
+ROSA_TOKEN=$(read_profile_file "ocm-token")
+if [[ -n "${SSO_CLIENT_ID}" && -n "${SSO_CLIENT_SECRET}" ]]; then
+  log "Logging into ${OCM_LOGIN_ENV} with SSO credentials using rosa cli"
+  rosa login --env "${OCM_LOGIN_ENV}" --client-id "${SSO_CLIENT_ID}" --client-secret "${SSO_CLIENT_SECRET}"
+  ocm login --url "${OCM_LOGIN_ENV}" --client-id "${SSO_CLIENT_ID}" --client-secret "${SSO_CLIENT_SECRET}"
+elif [[ -n "${ROSA_TOKEN}" ]]; then
+  log "Logging into ${OCM_LOGIN_ENV} with offline token using rosa cli"
+  rosa login --env "${OCM_LOGIN_ENV}" --token "${ROSA_TOKEN}"
+  ocm login --url "${OCM_LOGIN_ENV}" --token "${ROSA_TOKEN}"
+else
+  log "Cannot login! You need to securely supply SSO credentials or an ocm-token!"
+  exit 1
+fi
+
+
+if [[ -z "${UPGRADED_TO_VERSION}" ]]; then
+  log "Get the latest version!"
+  if [[ "$CHANNEL_GROUP" == "nightly" ]]; then
+    log "It doesn't support to upgrade with nightly version now"
+    exit 1
+  fi
+  # Get the latest OCP version
+  version_cmd="rosa list version --hosted-cp --channel-group $CHANNEL_GROUP -o json"
+  filter_cmd="$version_cmd | jq -r '.[] | .raw_id'"
+  versionList=$(eval $filter_cmd)
+  echo -e "Available cluster versions:\n${versionList}"
+  target_version=$(echo "$versionList" | head -1 || true)
+  UPGRADED_TO_VERSION=$target_version
+fi
+
+HOSTED_CP=$(rosa describe cluster -c $cluster_id -o json | jq -r '.hypershift.enabled')
+
+while true; do
```

Comment:
We need a timeout here. The timeout should be `total job running time on prow` - `cluster creation time` - `cluster deletion time`

Reply:
Added the timeout